pkg/apis/experimental/types.go

```diff
@@ -539,3 +539,32 @@ type IngressBackend struct {
 	// Specifies the protocol of the referenced service.
 	Protocol api.Protocol `json:"protocol,omitempty"`
 }
+
+type ConfigDataSelector struct {
```

Comment:
@pmorie I was modeling the implementation after downward API plug-in. I added that as a placeholder. What we'll do is refactor ``ObjectFieldSelector`` to select fields from arbitrary API objects. At the moment it selects fields from the pod in a hard-coded manner.